### PR TITLE
Add controller type information in containerPodData in kubeturbo discovery

### PR DESCRIPTION
**Pull request by libai98**
I have merged this PR https://github.com/turbonomic/kubeturbo/pull/863 in kubeturbo to update protobuf from go-sdk to create SLO action merge spec. This is the follow up PR for using controller type information in containerPodData for exclusion filter of certain containerPod from SLO merge action.

Developer Checks
[x] Full build with unit tests and Checkstyle / SpotBugs tests
[x] Unit tests added / updated
[x]No unlicensed images, no third-party code (such as from StackOverflow)
Integration (Robot Framework) tests added / updated
Manual testing done (and described)
Product sweep run and passed
Developer wiki updated (and linked to this description)

### CODE REVIEW

File: pkg/discovery/dtofactory/pod_entity_dto_builder.go [hunk 1]
```diff
@@ -596,6 +599,30 @@ func (builder *podEntityDTOBuilder) createContainerPodData(pod *api.Pod) *proto.
...
+		// Pod does not have controller
+		glog.V(3).Infof("Skip adding workload controller data for pod %v/%v: pod has no controller.",
+			pod.Namespace, pod.Name)
+		return podData
+	}
+
+	controller, exists := builder.ClusterSummary.ControllerMap[ownerInfo.Uid]
+	if !exists {
+		// Could be custom controller. We do not bulk process custom controller.
+		glog.V(3).Infof("add cumtomer controller data of controller %v/%v for pod %v/%v: controller not cached.",
```
Comment (irfanurrehman) [inline pkg/discovery/dtofactory/pod_entity_dto_builder.go:614]: Does this mean that we might update the type as custom controller for a replicaset or deployment parent if the same has not been cached yet??
Would using `GetPodControllerInfo()` (currently a method on clusterScraper) be better in this case.

**Review by irfanurrehman — COMMENTED** [review]
Thanks @libai98 for doing this.

File: pkg/discovery/dtofactory/pod_entity_dto_builder.go [hunk 2]
```diff
@@ -639,3 +666,65 @@ func AppendNewLabelCommodityToList(commoditiesList []*proto.CommodityDTO, key st
...
+//   - KindCronJob
+//   - KindDaemonSet
+//   - KindDeployment
+//   - KindJob
+//   - KindReplicaSet
+//   - KindReplicationController
+//   - KindStatefulSet
+//
+// If the provided kind does not match any known controller type, it returns nil.
+func CreateWorkloadControllerDataByControllerType(kind string) *proto.EntityDTO_WorkloadControllerData {
```
[the remark this answers is not in the document]
Comment (libai98) [reply pkg/discovery/dtofactory/pod_entity_dto_builder.go]: Thanks for the suggestion! I did some code refactoring for the common code.

File: pkg/discovery/dtofactory/pod_entity_dto_builder.go (hunk 1 above)
Comment (libai98) [reply pkg/discovery/dtofactory/pod_entity_dto_builder.go]: I changed the code to use GetPodControllerInfo() to get the controller type for the pod. Please check that out and help me review on the default case of CustomControllerData. Thanks for the help on this again!

**Review by irfanurrehman — APPROVED** [review]
Thanks for doing this @libai98

**Review by amd-ibm — APPROVED** [review]

**Pull request merged**